stdlib/http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -301,20 +300,14 @@ public static void populateEntityBody(Context context, BMap<String, BValue> http
                 && context != null) {
             MultipartDecoder.parseBody(context, entity, contentType, httpMessageDataStreamer.getInputStream());
         } else {
-            long contentLength = NO_CONTENT_LENGTH_FOUND;
-            String lengthStr = httpCarbonMessage.getHeader(HttpHeaderNames.CONTENT_LENGTH.toString());
-            try {
-                contentLength = lengthStr != null ? Long.parseLong(lengthStr) : contentLength;
-                if (contentLength == NO_CONTENT_LENGTH_FOUND) {
-                    //Read one byte to make sure the incoming stream has data
-                    contentLength = httpCarbonMessage.countMessageLengthTill(ONE_BYTE);
-                }
-            } catch (NumberFormatException e) {
-                throw new BallerinaException("Invalid content length");
-            }
+            long contentLength = MimeUtil.extractContentLength(httpCarbonMessage);
             if (contentLength > 0) {
-                entity.addNativeData(ENTITY_BYTE_CHANNEL, new EntityWrapper(
-                        new EntityBodyChannel(httpMessageDataStreamer.getInputStream())));
+                if (isStreaming) {
+                    entity.addNativeData(ENTITY_BYTE_CHANNEL, new EntityWrapper(
+                            new EntityBodyChannel(httpMessageDataStreamer.getInputStream())));
```

Comment:
Shall we initialize the httpMessageDataStreamer here and inside the multipart if block instead of creating it at the beginning of the function? Now that the design has been changed, a lot of the calls will go to the following if block, and we will end up creating httpMessageDataStreamer objects unnecessarily.